src/PluginForm/Bluesnap/HostedPaymentFieldsPaymentMethodAddForm.php

```diff
@@ -0,0 +1,214 @@
+<?php
+
+namespace Drupal\commerce_bluesnap\PluginForm\Bluesnap;
+
+use Drupal\commerce\InlineFormManager;
+use Drupal\commerce_bluesnap\ApiService;
+use Drupal\commerce_payment\PluginForm\PaymentMethodAddForm as BasePaymentMethodAddForm;
+
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Routing\RouteMatchInterface;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Adds a credit card payment method form for the HostedPaymentFields gateway.
+ *
+ * @package Drupal\commerce_bluesnap\PluginForm\Bluesnap
+ */
+class HostedPaymentFieldsPaymentMethodAddForm extends BasePaymentMethodAddForm {
+
+  /**
+   * The Bluesnap API helper service.
+   *
+   * @var \Drupal\commerce_bluesnap\ApiService
+   */
+  protected $apiService;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(
+    InlineFormManager $inline_form_manager,
+    RouteMatchInterface $route_match,
+    EntityTypeManagerInterface $entity_type_manager,
+    LoggerInterface $logger,
+    ApiService $api_service
+  ) {
+    parent::__construct(
+      $inline_form_manager,
+      $route_match,
+      $entity_type_manager,
+      $logger
+    );
+
+    $this->apiService = $api_service;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.commerce_inline_form'),
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager'),
+      $container->get('logger.factory')->get('commerce_payment'),
+      $container->get('commerce_bluesnap.api_service')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildCreditCardForm(
```

Comment:
**CHANGE REQUEST**

*API*

Looking at the parent class this is a `protected` method.